CMakeLists.txt

```diff
@@ -366,10 +364,6 @@ if(MSVC)
   set(US_CXX_FLAGS "/MP /WX ${US_CXX_FLAGS}")
 else()
 
-  if(US_COMPILER_APPLE_CLANG)
-    set(US_CXX_FLAGS "${US_CXX_FLAGS} -stdlib=libc++")
```

Comment:
In the current C++ MicroServices repo, this is replaced by:

```if (APPLE AND US_COMPILER_APPLE_CLANG)
  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++ -lc++abi")
  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -stdlib=libc++ -lc++abi")
  set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -stdlib=libc++ -lc++abi")
```
Everything in this PR builds/compiles/runs fine on Linux, but I don't have any way to test this with Mac. I assume this is fine, but just want to make sure we're good without this flag on Mac.

Reply:
I don't think that's the correct way to handle the library and ABI flags: it's completely dependent on the system's installation setup (some clang installations use libstdc++, some use libc++), and this is the incorrect way to set flags in CMake since those options only apply to subdirectories rather than downstream targets. It's not clear to me why microservices has to touch these flags at all; they should be set as a system/project-wide configuration by the user...